runtime-sdk/src/modules/rofl/app/registration.rs

```diff
@@ -143,6 +145,11 @@ where
             }
         };
 
+        // Include provider-specific metadata if available.
+        if let Err(err) = self.collect_provider_metadata(&mut metadata).await {
```

Comment:
[nit] Might be more readable if we renamed this function to `fetch_provider_metadata` and then insert it here, as currently the reader has no idea what is going on with metadata.